core/libp2p/transport/impl/transport_impl.cpp

```diff
@@ -5,32 +5,40 @@
 
 #include "libp2p/transport/impl/transport_impl.hpp"
 
-#include <boost/asio/ip/address.hpp>
-#include <boost/lexical_cast.hpp>
 #include "libp2p/multi/multiaddress.hpp"
 #include "libp2p/transport/impl/transport_listener_impl.hpp"
+#include "libp2p/transport/impl/transport_parser.hpp"
 #include "libp2p/transport/tcp/tcp_connection.hpp"
 
 namespace libp2p::transport {
-  using boost::asio::ip::make_address;
   using multi::Multiaddress;
   using multi::Protocol;
 
+  /**
+   * Visitor for std::variant that return the address data corresponding to the
```

Comment:
```suggestion
   * Visitor for std::variant that returns the address corresponding to the
```
Not sure about the `address data` formulation

Reply:
It returns some data structure with values of a parsed multiaddress. I think that calling it an address may be misleading.